src/test/java/com/google/cloud/solutions/autotokenize/pipeline/EncryptionPipelineIT.java

```diff
@@ -89,22 +87,16 @@
   private final String inputSchemaJsonFile;
   private final int expectedRecordsCount;
 
-  public static transient DB testDBInstance;
+  private static MySQLContainer databaseContainer;
```

Comment:
static on variable is required because it's used by the _tearDownTestDB_ method which must be static (Junit requires this for @AfterClass to work).
Another solution is to 'wrap' the databaseContainer in an ExternalResource and remove the tearDownTestDB at all. Something like this :
`@Rule
public ExternalResource resource = new ExternalResource() {
  @Override
  protected void after() {
    if (databaseContainer != null) {
      databaseContainer.stop();
    }
  }
};`
Do you find this solution a better one?

Reply:
FYI there is no need to implement `ExternalResource` - Testcontainers classes already implement JUnit's rule protocol, so it can be as simple as:
```
@ClassRule
public static MySQLContainer<?> databaseContainer = new MySQLContainer<>("mysql:8.0.24")
    .withDatabaseName(testDatabaseName)
    .withUsername("root")
    .withPassword("")
    .withInitScript(initScriptFile);
```

(no need to start / stop it manually)

Reply:
Many thanks @bsideup!
The problem is also that this test is a parametrized one and the database is used only by some cases/parameters, not all. Right now the database is started when it is really needed for a test case.
But I think it doesn't hurt to start the container earlier if we know for sure that a test case/parameter will use it.

Reply:
I will go ahead and use @ClassRule as you mentioned @bsideup.
Thanks again!

Reply:
@bsideup thanks for the suggestion.
@posabogdanpetre lets keep the structure for now. The DB tests can be separated later to ensure DB container is spinned up only when required.